xonsh/parser.py

```diff
@@ -1516,8 +1520,8 @@ def p_atom(self, p):
             bt = '`'
             if isinstance(p1, (ast.Num, ast.Str, ast.Bytes)):
                 pass
-            elif (p1 is True) or (p1 is False) or (p1 is None):
-                p1 = ast.NameConstant(value=p1, lineno=self.lineno,
+            elif (p1 == 'True') or (p1 == 'False') or (p1 == 'None'):
+                p1 = ast.NameConstant(value=eval(p1), lineno=self.lineno,
```

Comment:
It would be nice to split this out into three separate elif-statements and not eval a string that we know to be a singleton.  It is more lines of code, but likely faster overall.